BCStudentSoftwareDevTeam/lsf
# 425 inactive departments

Issue - There are some departments that we would consider inactive within our system. The definition for being inactive is that these departments are in our local database but they have been removed from the Tracy DB. As an enhancement, we wanted to split these inactive departments into two different tabs on the administrative "Manage Departments" page for ease of use.

Solution - We added a column, isActive, into the database for the department table. There is a script run once during the night on our application which is named updateStudentRecords.py which now checks for the most current info on all personnel and departments from Tracy. By checking once daily we ensure that the departments are correctly marked as active or inactive. In the route for the "Manage Departments" page, we simply query for the active departments and inactive departments separately before loading them into two different tabs on the page. 

To Test - Follow these steps:
1. Run `source setup.sh`
2. Reset your DB `database/reset_database.sh from-backup`
3. Change your environment to staging `export FLASK_ENV=staging`. This step ensures that the following script checks the real Tracy DB instead of the development one.
4. Run the updateStudentRecords.py script, `python scripts/updateDBRecords.py`. Be sure to wait until this completes.
5. Run the application `flask run`.
6. Navigate to the Manage Departments page by clicking Administration on the left-hand side of the window.

How you know it's good:
Ensure that there are departments in both the active and inactive departments tabs. Additionally, you should also ensure that the tabs function as you believe tabs should (shifting focus on click, that kind of thing). There should be some departments in both. For us, there were 143 active and 44 inactive departments. 

Resolves #425
